fiaas_deploy_daemon/deployer/kubernetes/service_account.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+# Copyright 2017-2019 The FIAAS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+
+import logging
+
+from k8s.client import NotFound
+from k8s.models.common import ObjectMeta, LocalObjectReference
+from k8s.models.service_account import ServiceAccount
+
+from fiaas_deploy_daemon.retry import retry_on_upsert_conflict
+from fiaas_deploy_daemon.tools import merge_dicts
+
+
+LOG = logging.getLogger(__name__)
+
+
+class ServiceAccountDeployer(object):
+    def __init__(self, config, owner_references):
+        self._owner_references = owner_references
+
+    def deploy(self, app_spec, labels):
+        self._create(app_spec, labels)
+
+    def delete(self, app_spec):
+        LOG.info("Deleting serviceAccount for %s", app_spec.name)
+        try:
+            service_account = ServiceAccount.get(app_spec.name, app_spec.namespace)
+            if not self._owned_by_fiaas(service_account):
+              return
+
+            ServiceAccount.delete(app_spec.name, app_spec.namespace)
+        except NotFound:
+            pass
+
+    @retry_on_upsert_conflict
+    def _create(self, app_spec, labels):
+        LOG.info("Creating/updating serviceAccount for %s with labels: %s", app_spec.name, labels)
+        metadata = ObjectMeta()
+        image_pull_secrets = []
+        automount_service_account_token = True
+        try:
+            service_account = ServiceAccount.get(app_spec.name, app_spec.namespace)
+            if not self._owned_by_fiaas(service_account):
+              return
+        except NotFound:
+            pass
+
+        default_service_account = ServiceAccount.get("default", app_spec.namespace)
+        image_pull_secrets = default_service_account.imagePullSecrets
+        service_account_name = app_spec.name
+        custom_labels = labels
+        custom_annotations = {}
+        if "service_account" in app_spec.labels:
```

Comment:
I think this isn't correct; app_spec.labels is a namedtuple, with properties, rather than a dict. You should update LabelAndAnnotationSpec in models.py to add the new property, and then populate it in _labels_annotations_spec inside v3/factory.py
(Then the 'if' becomes unnecessary here; the value will definitely be a dict, and will be mergeable even if it's empty)